File: paicord-rs/PaicordLibBridge/Sources/DiscordHTTP/Types/Permission.swift
import PaicordLib

extension RoleRust {
    init(role: Role) {
        self.id = SnowflakeRust(inner: UInt64(role.id.rawValue) ?? 0)
        //self.name = role.name.intoRustString()
        // self.description = role.description.map { $0.intoRustString() }
        self.color = DiscordColorRust(color: role.color)
        // self.hoist = role.hoist
        // self.icon = role.icon.map { $0.intoRustString() }
        // self.unicode_emoji = role.unicode_emoji.map { $0.intoRustString() }
        self.position = Int32(role.position)
        // self.managed = role.managed
        // self.mentionable = role.mentionable
        // self.version = role.version.map { Int32($0) }
    }
}

File: paicord-rs/src/discord_models/types/permission.rs
use crate::ffi;

pub type Role = ffi::RoleRust;

impl std::fmt::Debug for Role {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        f.debug_struct("RoleRust")
            .field("id", &self.id)
            // .field("name", &self.name)
            // .field("description", &self.description)
            .field("color", &self.color)
            // .field("hoist", &self.hoist)
            // .field("icon", &self.icon)
            // .field("unicode_emoji", &self.unicode_emoji)
            .field("position", &self.position)
            // .field("managed", &self.managed)
            // .field("mentionable", &self.mentionable)
            // .field("version", &self.version)
            .finish()
    }
}

unsafe impl Send for Role {}
unsafe impl Sync for Role {}